### PR TITLE
MVP implementation

### PR DESCRIPTION
**Pull request by szg251**
The MVP includes:
- type definitions
- PlutusData conversion traits
- optional Json conversion traits using LambdaBuffers

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+              cargoExtraArgs = "--features lbf";
+            });
+
+            checks."${crateName}-clippy" = craneLib.cargoClippy (commonArgs // {
+              inherit cargoArtifacts;
+            });
+          };
+      };
+
+}
```
[the remark this answers is not in the document]
Comment (bladyjoker) [reply flake.nix:67]: Aha, so the default setup just works! Marvelous!

**Review by bladyjoker — APPROVED** [review]
This is gorgeous! or might one say Gergeous?

File: flake.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (szg251) [reply flake.nix:67]: exactly

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            };
+
+            devShells.default = craneLib.devShell { checks = self'.checks; };
+
+            packages.default = craneLib.buildPackage (commonArgs // {
+              inherit cargoArtifacts;
+              doTest = false;
+            });
+
+            pre-commit.settings.hooks.rustfmt.enable = true;
```
[the remark this answers is not in the document]
Comment (szg251) [reply flake.nix:54]: good point, I'll add these

**Pull request merged**